File: juste_prix.py
#Python version

import random

prix_random = random.randint(1, 100)
print(prix_random)
tentatives = 0
print("Devinez un nombre entre 1 et 100 !")
while True:
    tentatives += 1
    try:
        prix_user = int(input(f"Tentative {tentatives} - Quel prix ? "))
        if prix_user == prix_random:
            print(f"Bien joué ! Vous avez trouvé en {tentatives} tentative(s) !")
            break
        elif prix_user < prix_random:
            print("C'est plus")
        else:
            print("C'est moins")   
    except ValueError:
        print("Erreur : Veuillez entrer un nombre valide ! (Cette tentative compte)")
        continue

File: juste_prix.js
// JavaScript version

let prix_random = Math.floor(Math.random() * 100) + 1;
console.log(prix_random);
let tentatives = 0;
console.log("Devinez un nombre entre 1 et 100 !");
while (true) {
    tentatives++;
    let user_price = prompt(`Tentative ${tentatives} - Quel prix ?`);
    let user_price_int = parseInt(user_price);
    if (isNaN(user_price_int)) {
        console.log("Erreur : Veuillez entrer un nombre valide ! (Cette tentative compte)");
        continue;
    }
    if (user_price_int == prix_random) {
        console.log(`Bien joué ! Vous avez trouvé en ${tentatives} tentative(s) !`);
        break;
    } else if (user_price_int < prix_random) {
        console.log("C'est plus");
    } else {
        console.log("C'est moins");
    }
}
